src/tree/v2/CompatibleBranchDataItem.ts

```diff
@@ -0,0 +1,62 @@
+import { AzExtTreeItem, CompatibleQuickPickOptions, CreateOptions, isAzExtParentTreeItem } from '@microsoft/vscode-azext-utils';
+import * as vscode from 'vscode';
+import { localize } from 'vscode-nls';
+import { ApplicationResource, BranchDataProvider, ResourceModelBase } from '../../api/v2/v2AzureResourcesApi';
+import { BranchDataItem, BranchDataItemOptions } from './BranchDataItem';
+import { createBranchDataItemFactory } from './factories/branchDataItemFactory';
+import { ResourceGroupsItem } from './ResourceGroupsItem';
+import { ResourceGroupsItemCache } from './ResourceGroupsItemCache';
+
+export class CompatibleBranchDataItem extends BranchDataItem {
+    constructor(
+        branchItem: ResourceModelBase,
+        branchDataProvider: BranchDataProvider<ApplicationResource, ResourceModelBase>,
+        itemCache: ResourceGroupsItemCache,
+        options: BranchDataItemOptions | undefined
+    ) {
+        super(branchItem, branchDataProvider, itemCache, options);
+        itemCache.addBranchItem(this.branchItem, this);
+    }
+
+    async getChildren(): Promise<ResourceGroupsItem[] | undefined> {
+        const children = await this.branchDataProvider.getChildren(this.branchItem);
+        const factory = createBranchDataItemFactory(this.itemCache);
+        return children?.map(child => factory(child, this.branchDataProvider));
+    }
+
+    public get quickPickOptions(): CompatibleQuickPickOptions {
```

Comment:
This can be removed now.

Reply:
I *think* that's removed in the next PR https://github.com/microsoft/vscode-azureresourcegroups/pull/382